pkgs/by-name/gn/gnome-maps/package.nix

```
@@ -97,6 +96,12 @@ stdenv.mkDerivation (finalAttrs: {
     rm $out/lib/gnome-maps/libgnome-maps.so.0
   '';
 
+  preFixup = ''
+    substituteInPlace "$out/share/applications/org.gnome.Maps.desktop" \
+      --replace-fail "Exec=gapplication launch org.gnome.Maps" \
+                     "Exec=${lib.getExe' glib "gapplication"} launch org.gnome.Maps"
```

```suggestion
                     "Exec=org.gnome.Maps"
```

Why is this not working?

---

The thing is, it probably also does work.

I didn't think of the option to just write `"Exec=gnome-maps"` because I was too focused on full paths.

I decided to first go with only replacing gapplication with it's full path to directly address issue #53111's title `Some packages do not use absolute gapplication path in desktop files`.

But as I said I am open to changing it to `Exec=gnome-maps` if gnome maintainers don't say anything against it.

I just noticed that `gnome-clocks` has exactly this in it's .desktop file so I guess they won't mind.

---

> I didn't think of the option to just write `"Exec=gnome-maps"` because I was too focused on full paths.

Well I'm also too focused on removing full paths, see https://github.com/NixOS/nixpkgs/issues/308324

---

Didn't know about that, thank you for mentioning this.